performant-software/Neatline
# Tighten WKT formatting check in backend

### What does this PR do?
- Processes a coverage strings through geoPHP in order to catch formatting errors before they cause save failure
- Prevents Neatline records with coverage imported from an Omeka item from creating a map marker that won't persist

### What issues does it address?
Closes #388, closes #392

### How to test
- On [staging](http://neatline-staging.herokuapp.com/), open the editor for a new or existing Neatline exhibit.
- Add a new record and click the Item tab to select an Omeka item from the list.
 Choose an item that has been given a malformed WKT string as its coverage field,
 e.g. the "WKT Test" item current has the coverage field `(GEOMETRYCOLLECTION(POINT(-60626.80528947, 6661916.076664))`, which contains an extraneous comma
- Click save after selecting the item; the record should save successfully, but no map marker should appear. Clicking on the Map tab should show no value in the Coverage field.
- Repeat the process with an item that has correctly formatted WKT in the coverage field (e.g. remove the comma from the WKT Test item's coverage value). The map marker should appear upon save. Closing the record should cause the map marker to momentarily disappear but then reappear as map layer are refreshed.
